.github/workflows/build.yml

```diff
@@ -11,18 +11,37 @@ jobs:
     name: Goreleaser Dry
     runs-on: ubuntu-latest
     steps:
+      - name: Check out the code
+        uses: actions/checkout@v3
+
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
           go-version: ^1.18
 
-      - name: Install llvm
-        run: sudo apt-get install llvm libelf-dev
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          components: rustfmt, clippy
 
-      - name: Checkout
-        uses: actions/checkout@v3
+      - name: Set up Rust Nightly
+        uses: actions-rs/toolchain@v1
         with:
-          fetch-depth: 0
+          profile: minimal
+          toolchain: nightly
+
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@v1.3.0
+
+      - name: Run eBPF toolchain setup
+        run: |
+          make -C bpf setup
+          cd bpf && cargo check
+
+      - name: Install llvm
+        run: sudo apt-get update -y && sudo apt-get install llvm libelf-dev
```

Comment:
nit: We are also installing `libelf-dev`, should we change the name so it's more descriptive?